mypy/plugins/default.py

```diff
@@ -238,37 +238,65 @@ def typed_dict_get_signature_callback(ctx: MethodSigContext) -> CallableType:
 
 def typed_dict_get_callback(ctx: MethodContext) -> Type:
     """Infer a precise return type for TypedDict.get with literal first argument."""
-    if (isinstance(ctx.type, TypedDictType)
-            and len(ctx.arg_types) >= 1
-            and len(ctx.arg_types[0]) == 1):
-        keys = try_getting_str_literals(ctx.args[0][0], ctx.arg_types[0][0])
-        if keys is None:
-            return ctx.default_return_type
+    if not (
+        isinstance(ctx.type, TypedDictType)
+        and len(ctx.arg_types) >= 1
+        and len(ctx.arg_types[0]) == 1
+    ):
+        return ctx.default_return_type
 
-        output_types: List[Type] = []
-        for key in keys:
-            value_type = get_proper_type(ctx.type.items.get(key))
-            if value_type is None:
-                return ctx.default_return_type
-
-            if len(ctx.arg_types) == 1:
-                output_types.append(value_type)
-            elif (len(ctx.arg_types) == 2 and len(ctx.arg_types[1]) == 1
-                  and len(ctx.args[1]) == 1):
-                default_arg = ctx.args[1][0]
-                if (isinstance(default_arg, DictExpr) and len(default_arg.items) == 0
-                        and isinstance(value_type, TypedDictType)):
-                    # Special case '{}' as the default for a typed dict type.
-                    output_types.append(value_type.copy_modified(required_keys=set()))
-                else:
-                    output_types.append(value_type)
-                    output_types.append(ctx.arg_types[1][0])
-
-        if len(ctx.arg_types) == 1:
-            output_types.append(NoneType())
-
-        return make_simplified_union(output_types)
-    return ctx.default_return_type
+    keys = try_getting_str_literals(ctx.args[0][0], ctx.arg_types[0][0])
+    if keys is None:
+        return ctx.default_return_type
+
+    default_type: Optional[Type]
+    if len(ctx.arg_types) == 1:
+        default_type = None
```

Comment:
Seems like you could simplify by making this `default_type = NoneType()` here:
* You can then simplify the type of `default_type: Optional[Type]` to `default_type: Type`.
* You can then simplify the `default_type or NoneType()` expression later to just `default_type`.
* You can (probably) then eliminate the special `if default_type is None:` block.
   * However after eliminating that block you might need an extra length check in the following block: `if len(ctx.args[1]) == 1:`, such as `if len(ctx.args) >= 2 and len(ctx.args[1]) == 1`.